gemini/use-cases/retrieval-augmented-generation/utils/intro_multimodal_rag_utils.py

```diff
@@ -12,21 +12,21 @@
 from google.protobuf import struct_pb2
 from IPython.display import display
 from vertexai.language_models import TextEmbeddingModel
+from vertexai.vision_models import MultiModalEmbeddingModel, Image as vision_model_Image
 from vertexai.generative_models import (
     Image,
     HarmCategory,
     HarmBlockThreshold,
+    GenerationConfig
 )
 import PIL
 
 
 text_embedding_model = TextEmbeddingModel.from_pretrained("textembedding-gecko@latest")
+multimodal_embedding_model = MultiModalEmbeddingModel.from_pretrained("multimodalembedding@001")
 
-# textembedding-gecko@latest [Always pulls the latest]
-# textembedding-gecko@003 [Latest as on 14Feb 2024]
-
-# Function for getting text and image embeddings
 
+# Functions for getting text and image embeddings
 
 def get_text_embedding_from_text_embedding_model(
```

Comment:
Question on this - Why are the text embeddings being calculated with a different model than the images? Don't all of the embeddings need to be made with the same model for them to match correctly?

Reply:
Text embedding from the multimodal embeddings has a limit of 32 tokens (from [doc](https://cloud.google.com/vertex-ai/docs/generative-ai/embeddings/get-multimodal-embeddings#api-limits): The maximum text length is 32 tokens (approximately 32 words). If the input exceeds 32 tokens, the model internally shortens the input to this length.) 

The image description extracted from Gemini, my primary source for finding information in the images and later using it to match with other page text, is usually greater than 32 tokens. Hence, I use two different vector dimensions for pure text-text match and image-image match.

Reply:
I still think it would make sense to use the text and images together when making the embeddings, maybe you can limit the token output for the gemini pro vision descriptions so they will fit?

Won't block this PR for now, but something to think on.